app/notifications/validators.py

```diff
@@ -274,17 +275,62 @@ def check_service_letter_contact_id(service_id, letter_contact_id, notification_
             raise BadRequestError(message=message)
 
 
-def decode_personalisation_files(personalisation_data):
+def validate_personalisation_and_decode_files(json_personalisation):
     errors = []
-    file_keys = [k for k, v in personalisation_data.items() if isinstance(v, dict) and "file" in v]
+    json_personalisation, errors_vars = validate_personalisation(json_personalisation)
+    json_personalisation, errors_num_file = validate_personalisation_num_files(json_personalisation)
+    json_personalisation, errors_files = decode_personalisation_files(json_personalisation)
+    errors.extend(errors_vars)
+    errors.extend(errors_num_file)
+    errors.extend(errors_files)
+    return json_personalisation, errors
+
+
+def validate_personalisation(json_personalisation):
+    errors = []
+    values = [v for _, v in json_personalisation.items() if not isinstance(v, dict)]
+    concat_values = functools.reduce(lambda v1, v2: v1 + v2, values, "")
```

Comment:
these values are almost certainly strings, I think... just wondering if we should make sure, like returning `f"{v1}{v2}"` or something.

Reply:
Yeah these are strings, string interpolation could work (in the lambda), but I'd still need to reduce the values.

Reply:
💯  Was thinking along the lines of
```
    concat_values = functools.reduce(lambda v1, v2: f"{v1}{v2}", values, "")
```